pkg/ebpf/hooked_syscall_table.go

```diff
@@ -0,0 +1,106 @@
+package ebpf
+
+import (
+	gocontext "context"
+	"runtime"
+	"strings"
+	"time"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/derive"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/utils"
+)
+
+const syscallPrefix = "__x64_sys_"
+
+var expectedSyscallTableInit = false
+
+// hookedSyscallTableRoutine the main routine that checks if there's a hooked syscall in the syscall table.
+// It runs on tracee's startup and from time to time.
+func (t *Tracee) hookedSyscallTableRoutine(ctx gocontext.Context) {
+	logger.Debugw("Starting hookedSyscallTable goroutine")
+	defer logger.Debugw("Stopped hookedSyscallTable goroutine")
+
+	if t.eventsState[events.HookedSyscall].Emit == 0 {
+		return
+	}
+
+	if runtime.GOARCH != "amd64" {
+		logger.Debugw("hookedSyscallTable integrity checker is only supported on amd64")
+		return
+	}
```

Comment:
This is a no go for me. The same mechanism would work in arm64 (The only arch we support) and `core_arm64.go` could be changed in the same or in a similar way.